module/VuFind/src/VuFind/RecordDriver/SolrOverdrive.php

```diff
@@ -361,13 +374,15 @@ public function isCheckedOut()
      */
     public function isHeld()
     {
-        $overDriveId = $this->getOverdriveID();
-        $result = $this->connector->getHolds(true);
-        if ($result->status) {
-            $holds = $result->data;
-            foreach ($holds as $hold) {
-                if (strtolower($hold->reserveId) == $overDriveId) {
-                    return $hold;
+        if ($this->isLoggedIn()) {
```

Comment:
In #3459, I eliminated the isLoggedIn method that returns a user object, since it's a bit confusing to have a method that sounds like a Boolean but returns a non-Boolean value. While working on that, I noticed that Overdrive had an isLoggedIn method of its own, but that method is not used at all in the dev branch. I see that you _are_ using it here. I'd suggest either renaming the method to something like `getCatalogUser` if you want it to return an object, or changing it to return a Boolean value (since that seems to be the only way you're using it anyway).

Reply:
I changed the return type. But let me know if there is a better way to do it. I couldn't find any other drivers that checks for a user so maybe there is a more fundamental problem here.

Reply:
I think this is fine for now at least!